pkg/resources/task_state_upgraders.go

```diff
@@ -0,0 +1,49 @@
+package resources
+
+import (
+	"context"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+)
+
+func v098TaskStateUpgrader(ctx context.Context, rawState map[string]any, meta any) (map[string]any, error) {
+	if rawState == nil {
+		return rawState, nil
+	}
+
+	rawState["condition"] = rawState["when"]
+	rawState["started"] = booleanStringFromBool(rawState["enabled"].(bool))
+	rawState["allow_overlapping_execution"] = booleanStringFromBool(rawState["allow_overlapping_execution"].(bool))
+	if rawState["after"] != nil {
```

Comment:
what if after was not an array for some reason, should we check it before getting len?

Reply:
Hmm I checked and first version of the tasks (4 years old) had string `after` I guess it depends on how far do we want to look. I'll add explicit check.